src/java.base/share/classes/java/security/SecureRandom.java

```diff
@@ -732,8 +732,11 @@ public void setSeed(long seed) {
         /*
          * Ignore call from super constructor (as well as any other calls
          * unfortunate enough to be passing 0).  It's critical that we
-         * ignore call from superclass constructor, as digest has not
-         * yet been initialized at that point.
+         * ignore call from superclass constructor. All SecureRandom
+         * constructors call `super(0)` but we do not want `setSeed(0)`
+         * to actually do any seeding. We either keep the object unseeded
+         * (in `new SecureRandom()`) or we seed the object explicitly
+         * (in `new SecureRandom(byte[])`).
```

Comment:
Maybe clearer to make it shorter? I removed the "It's critical..." sentence as it seems redundant.
/*
 * Ignore call from super constructor as well as any other calls 
 * unfortunate enough to be passing 0. All SecureRandom
 * constructors call `super(0)` which leads to `setSeed(0)`.
 * We either keep the object unseeded (in `new SecureRandom()`) 
 * or we seed the object explicitly (in `new SecureRandom(byte[])`).
 */

Reply:
Sure I can do that.